app/services/GetMessagesService.scala

```diff
@@ -30,30 +30,47 @@ import scala.concurrent.{ExecutionContext, Future}
 class GetMessagesService @Inject()(
                                     messagesConnector: GetMessagesConnector,
                                     markMessageAsReadConnector: MarkMessageAsReadConnector,
+                                    getSubmissionFailureMessageConnector: GetSubmissionFailureMessageConnector,
                                     messageInboxRepository: MessageInboxRepository
                                   )(implicit ec: ExecutionContext) {
 
   def getMessages(ern: String, search: Option[MessagesSearchOptions])(implicit hc: HeaderCarrier): Future[GetMessagesResponse] = {
     messagesConnector.getMessages(ern, search).map {
       case Right(messages) =>
-        messages.messages.foreach { message =>
-          messageInboxRepository.set(MessageCache(ern, message))
-        }
+        messages.messages.foreach(message => messageInboxRepository.set(MessageCache(ern, message, errorMessage = None)))
         messages
       case _ => throw MessagesException(s"Error occurred when fetching messages for trader $ern")
     }
   }
 
   def getMessage(ern: String, uniqueMessageIdentifier: Long)(implicit hc: HeaderCarrier): Future[Option[MessageCache]] = {
 
-    messageInboxRepository.get(ern, uniqueMessageIdentifier).map {
-      case Some(message) =>
-        markMessageAsReadConnector.markMessageAsRead(ern, uniqueMessageIdentifier)
-        Some(message)
+    messageInboxRepository.get(ern, uniqueMessageIdentifier).flatMap {
+      case Some(cachedMessage) =>
+        retrieveSubmissionFailureMessageIfMessageIs704(cachedMessage).map {
+          cachedMessageWithPossible704 =>
+            markMessageAsReadConnector.markMessageAsRead(ern, uniqueMessageIdentifier)
+            Some(cachedMessageWithPossible704)
+        }
       case _ =>
-        None
+        Future(None)
```

Comment:
Did you want to change this to Future.successful ?

Reply:
Will do it in another PR